File: RxUI/src/main/java/com/tamsiree/rxui/view/colorpicker/builder/ColorWheelRendererBuilder.java
package com.tamsiree.rxui.view.colorpicker.builder;


import com.tamsiree.rxui.view.colorpicker.ColorPickerView;
import com.tamsiree.rxui.view.colorpicker.renderer.ColorWheelRenderer;
import com.tamsiree.rxui.view.colorpicker.renderer.FlowerColorWheelRenderer;
import com.tamsiree.rxui.view.colorpicker.renderer.SimpleColorWheelRenderer;

/**
 * @author tamsiree
 * @date 2018/6/11 11:36:40 整合修改
 */
public class ColorWheelRendererBuilder {
    public static ColorWheelRenderer getRenderer(ColorPickerView.WHEEL_TYPE wheelType) {
        switch (wheelType) {
            case CIRCLE:
                return new SimpleColorWheelRenderer();
            case FLOWER:
                return new FlowerColorWheelRenderer();
                default:
                    break;
        }
        throw new IllegalArgumentException("wrong WHEEL_TYPE");
    }
}

File: RxUI/src/main/java/com/tamsiree/rxui/view/colorpicker/builder/ColorWheelRendererBuilder.kt
package com.tamsiree.rxui.view.colorpicker.builder

import com.tamsiree.rxui.view.colorpicker.ColorPickerView.WHEEL_TYPE
import com.tamsiree.rxui.view.colorpicker.renderer.ColorWheelRenderer
import com.tamsiree.rxui.view.colorpicker.renderer.FlowerColorWheelRenderer
import com.tamsiree.rxui.view.colorpicker.renderer.SimpleColorWheelRenderer

/**
 * @author tamsiree
 * @date 2018/6/11 11:36:40 整合修改
 */
object ColorWheelRendererBuilder {
    @JvmStatic
    fun getRenderer(wheelType: WHEEL_TYPE?): ColorWheelRenderer {
        when (wheelType) {
            WHEEL_TYPE.CIRCLE -> return SimpleColorWheelRenderer()
            WHEEL_TYPE.FLOWER -> return FlowerColorWheelRenderer()
            else -> {
            }
        }
        throw IllegalArgumentException("wrong WHEEL_TYPE")
    }
}